tox.ini

```diff
@@ -14,7 +14,8 @@ commands =
     m8a-fetch-language-containers {envname}
     # -n auto tells pytest to use as many processors as detected
     # to run all the tests from the module test_{envname} in parallel
-    pytest -n auto test_{envname}.py --junitxml={toxinidir}/junit.xml []
+    pytest -n auto -m "not serial" test_{envname}.py --junitxml={toxinidir}/junit_parallel.xml []
+    pytest -n0 -m "serial" test_{envname}.py --junitxml={toxinidir}/junit_serial.xml []
```

Comment:
You can remove -n0 . I think it's clearer, and that way the pytest-xdist isn't involved at all, iirc.

Reply:
Meh, let's ignore this ^